### PR TITLE
Support Test::Unit::TestCase

### PR DESCRIPTION
**Pull request by janosch-x**

### CODE REVIEW

File: lib/minitest_to_rspec/input/model/klass.rb [hunk 1]
```diff
@@ -78,23 +82,24 @@ def test_case?
...
+          exp.sexp_body.flat_map { |entry| ancestor_names(entry) }
         end
 
         def lineage?(exp, names)
           assert_sexp_type(:colon2, exp)
-          exp.length == names.length + 1 &&
-            names.each_with_index.all? { |name, ix|
-              name.to_sym == ancestor_name(exp, ix).to_sym
-            }
+          ancestor_names(exp) == names
```
Comment (jaredbeck) [inline lib/minitest_to_rspec/input/model/klass.rb]: The extraction of `ancestor_names` is a nice improvement. I like how it separates calculation from decision.

File: lib/minitest_to_rspec/input/model/klass.rb [hunk 2]
```diff
@@ -78,23 +82,24 @@ def test_case?
...
-          assert_sexp_type(:colon2, exp)
-          ancestor = exp[index + 1]
-          sexp_type?(:const, ancestor) ? ancestor[1] : ancestor
+        def ancestor_names(exp)
+          return [exp] if exp.is_a?(Symbol)
+
+          sexp_type?(:colon2, exp) || sexp_type?(:const, exp) ||
+            raise(TypeError, "Expected :const or :colon2, got #{exp.inspect}")
+
+          exp.sexp_body.flat_map { |entry| ancestor_names(entry) }
```
Comment (jaredbeck) [inline lib/minitest_to_rspec/input/model/klass.rb]: Normally I'm not a fan of recursion, but I think it looks great here. I tried writing an iterative version and it was quite ugly.

**Comment by jaredbeck** [comment]
> Maybe this would make the ModuleShorthandError obsolete?

I don't remember why it doesn't support shorthand. I personally only used nested-style, so it's not a feature I need, but I'd be happy to accept a PR.

**Comment by janosch-x** [comment]
> I don't remember why it doesn't support shorthand.

I didn't understand either. My tests had lots of shorthands, including in test class names, and the error never triggered. So I don't know what to do with this and just wanted to let you know that `#ancestor_names` *could* be re-used here.

**Pull request merged**